File: transactions/src/main/java/com/example/repository/TransactionRepository.java
package com.example.repository;

import com.example.model.Transaction;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import java.math.BigDecimal;
import java.util.List;

public interface TransactionRepository extends JpaRepository<Transaction, Long> {
    List<Transaction> findByAccountId(Long accountId);

    @Query("SELECT COALESCE(SUM(t.amount), 0) FROM Transaction t WHERE t.account.id = :accountId")
    BigDecimal sumAmountsByAccountId(@Param("accountId") Long accountId);
}

File: transactions/src/main/kotlin/com/example/repository/TransactionRepository.kt
package com.example.repository

import com.example.model.Transaction
import kotlinx.coroutines.flow.Flow
import org.springframework.data.r2dbc.repository.Query
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import reactor.core.publisher.Mono
import java.math.BigDecimal

interface TransactionRepository : CoroutineCrudRepository<Transaction, Long> {
    @Query("SELECT id, account_id, amount FROM transaction WHERE account_id = :accountId")
    fun findByAccountId(accountId: Long): Flow<Transaction>

    @Query("SELECT COALESCE(SUM(amount), 0) FROM transaction WHERE account_id = :accountId")
    fun sumAmountsByAccountId(accountId: Long): Mono<BigDecimal>
}
